gigaword-indexer/src/main/java/edu/isi/vista/gigawordIndexer/ConcatenatedGigawordDocuments.java

```diff
@@ -18,18 +26,112 @@
 
 public class ConcatenatedGigawordDocuments implements Iterable<ConcatenatedGigawordDocuments.Article> {
   private final String concatenatedFileText;
+  private List<Article> articleList;
+  boolean annotated;
 
   private ConcatenatedGigawordDocuments(String concatenatedFileText) {
     this.concatenatedFileText = Objects.requireNonNull(concatenatedFileText);
+    this.annotated = false;
   }
 
+  private ConcatenatedGigawordDocuments(String concatenatedFileText, List<Article> aArticleList) {
+    this.concatenatedFileText = Objects.requireNonNull(concatenatedFileText);
+    this.articleList = aArticleList;
+    this.annotated = true;
+  }
+  
   public static ConcatenatedGigawordDocuments fromGigwordGZippedFile(Path p) throws IOException {
     return new ConcatenatedGigawordDocuments(GZIPByteSource.fromCompressed(p.toFile())
             .asCharSource(StandardCharsets.UTF_8).read());
   }
 
+  public static ConcatenatedGigawordDocuments fromAnnotatedGigwordGZippedFile(Path p) throws IOException {
```

Comment:
I haven't yet done a full review pass, but wouldn't it be cleaner to do this as a separate class (e.g. `ConcatenatedAn notatedGigawordDocuments`) rather than within `ConcatenatedGigawordDocuments`?  The two code paths don't seem to share anything significant.

Reply:
Also, is the code below trying to maintain a correspondence between the offsets of the returned document text and the original source Gigaword document?

Reply:
@rgabbard As far as I could tell there is no way to do that. The Annotated Gigaword does not seem to include a "plain text" version of the data. The text comes in bracketed PTB format. We have no idea where whitespace was in the original Gigaword.

Reply:
@az79nefy can you split it out into another class?

Reply:
@reckart : I don't have Annotated Gigaword downloaded, but from the sample of the LDC catalog it looks like offset information is stored on tokens using `CharacterOffsetBegin` and `CharacterOffsetEnd`:
![LDC sample](https://catalog.ldc.upenn.edu/desc/addenda/LDC2012T21.jpg)

Using those you could construct a text to index which corresponds to the original text offsets perfectly for all tokens, which is all you care about.  

The reason to do this is that otherwise annotations become difficult to distribute (since the original text cannot even be transferred from one institution to another, much less posted publicly). But if everything is in terms of the original offsets, everyone with an LDC membership can just download Gigaword and apply the annotations.

Reply:
Split `ConcatenatedAnnotatedGigawordDocument` into new class in new commit.